File: 30804.py
import sys

n = int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))

temp = {}
left = 0
ans = 0 
for right in range(n):
    x = arr[right]

    if x in temp:
        temp[x] += 1
    else:
        temp[x] = 1

    while len(temp) > 2:
        y = arr[left]
        temp[y] -= 1

        if temp[y] == 0:
            del temp[y]
        
        left += 1

    ans = max(ans, right - left + 1)

print(ans)

File: 30804.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

int main(){
    int n;
    cin >> n;

    vector<int> arr(n, 0);

    for (int i=0; i<n; i++){
        cin >> arr[i];
    }
    
    int left = 0;
    int ans = 0;
    unordered_map<int ,int> check;

    for (int right=0; right<n; right++){
        check[arr[right]]++;

        while (check.size()>2){
            check[arr[left]]--;

            if (check[arr[left]] == 0){
                check.erase(arr[left]);
            }
            left++;
        }

        ans = max(ans, right-left+1);
    }

    cout << ans;
    return 0;
}